### ISSUE TITLE
Add 'unmanagedAcePathsRegex' for authorizables

### ISSUE THREAD

**Issue by jon-entwistle** [Netcentric/accesscontroltool#212]
ACEs added for groups via the web console are overwritten when accesstool is applied, requiring manual intervention to add these permissions again via the web console. 

The proposed feature allows paths to be configured at group level where existing privileges are respected when accesstool is run. These paths need to be recursive, so any ACEs for the group on subpaths are also respected.

An example group configuration would be:

    - A_GRP:
       - name: aName
         memberOf: anOther
         path: u
         honorPrivilege:
           - /respected/path/a
           - /respected/path/b

**Comment by ghenzler** [comment]
@jon-entwistle what about calling the property `restrictAcesToPaths`? That way the following could work (trying to give a more relevant example where it actually makes sense, most of the time we would exactly avoid such a path restriction as this can lead to strange effects):

```
- everyone:
   - name: Everyone
     restrictAcesToPaths:
       - /content/myproj
       - /content/dam/myproj
```
The above example would not touch the other out of the box ACEs for everyone, it would only add (or remove obsolete ones) the ones under the project root paths.

@jon-entwistle do you have maybe even a better example?

**Comment by jon-entwistle** [comment]
@ghenzler the name restrictAcesToPaths would imply that only ACEs which are on the given paths would be applied. This is not the case.

What honorPrivilege does is to find any ACEs which have been defined for the group prior to running the tool, and ensures that these are applied after the tool is run. There should be no side effects. It has exactly the same effect as running the tool (this resetting the permissions) then applying the ACEs which are not encapsulated by the .yaml by hand via the console. 

Hence the name, as any ACEs defined for the given group for the given paths will be honoured.

**Comment by ghenzler** [comment]
@jon-entwistle as discussed offline the "honor privileges" use case can be achieved differently (using https://github.com/Netcentric/accesscontroltool/blob/develop/docs/AdvancedFeatures.md#configure-memberships-to-dynamic-groups). But I think it would be reasonable to have a `restrictAcesToPaths` feature to allow ootb groups to be redefined without having to repeat all ACEs that come ootb - that way it would be easy to just add a few ACEs for e.g. `everyone` to certain repo locations without touching the ootb ACEs (see example in https://github.com/Netcentric/accesscontroltool/issues/212#issuecomment-316667053)

**Comment by inedoviesov1** [comment]
One more use case related to requested feature.

We have a group `approver` and we want to set permissions to AEM tools, workflows, some content paths via ACTool. But some part of the content might have permissions set manually, e.g. give `write` rights on new product in `/content/myapp/catalogue/new_product`.

In that case ACTool should be configurable in the way, that it ignores ACEs set it certain path and it would look like this:

```
- approver:
   - name: Approver
     ignoreAcesInPaths:
       - /content/myapp/catalogue
```

In this case we have to configure only the paths that have to have the same ACEs as before running ACTool, does not matter how they were set.

**Comment by mtstv** [comment]
@inedoviesov1 Why don't you want to create a new group "approver-custom", set permissions for that group manually and make your initial group "approver" as member of "approver-custom" inside of YAMl config?
P:S: "approver-custom" must not be part of YAML config.
       Only "approver" group must be configurated inside of YAML config.

**Comment by inedoviesov1** [comment]
@mtstv The point is that I would like to create groups in YAML config and then set permissions for these groups manually/programmatically.

**Comment by mtstv** [comment]
@inedoviesov1 Hi, just from my experience it is bad idea to set the rights manually after you set them programmatically. You will not have overview of that you have in Repo after that. And if you will involve manually over Useradmin GUI you can break the existing permissions (CRXDE must be used instead). Also in case of AEM disaster/migration/tests/content deletion/content copy you can't quickly restore the manually created permissions on the new instance in case of many of such permissions set manually. Also the Operations team usually don't have experience to do so.
Just tried to avoid to set permissions manually.

May be it is better in case of a new "/content/myapp/catalogue/new_product" to deploy a new YAML file  as a package and run AC Tool for selected JCR Path. It is simply for Operations team to do so. Also the set of YAML configs will be complete. Also it is possible to LOOP over the content nodes to set the permissions. may be it is also possible for your case.

**Comment by ghenzler** [comment]
After some discussion we intend to use `unmanagedAcePathsRegex` as property name (see https://github.com/Netcentric/accesscontroltool/wiki/Options-for-%22unmanaging%22-certain-aspects-of-configuration)

**Comment by ghenzler** [comment]
@mtstv I share your concerns. The ruleset on how the permissions are created for certain nodes in our particular project are just not easily transferred to a declarative YAML configuration.

**Comment by ghenzler** [comment]
Fixed with #247
